STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.cs

```diff
@@ -40,19 +42,35 @@ internal HttpExchangeService(IHttpBroker httpClientBroker)
                 httpExchange,
                 httpResponseMessage);
         });
-
+        
         private static HttpRequestMessage MapToHttpRequest(
-            HttpExchangeRequest httpExchangeRequest)
+            HttpExchangeRequest httpExchangeRequest,
+            HttpMethod defaultHttpMethod)
         {
             Uri baseAddress = new Uri(httpExchangeRequest.BaseAddress);
             string relativeUrl = httpExchangeRequest.RelativeUrl;
-
+            var httpMethod = GetHttpMethod(httpExchangeRequest, defaultHttpMethod);
+            
             return new HttpRequestMessage
             {
-                RequestUri = new Uri(baseAddress, relativeUrl)
+                RequestUri = new Uri(baseAddress, relativeUrl),
+                Version = new Version(httpExchangeRequest.Version),
+                Method = httpMethod
             };
         }
+        
+        private static HttpMethod GetHttpMethod(
```

Comment:
maybe instead HttpExchangeRequest httpExchangeRequest could be string string customHttpMethod

Reply:
Treated